test/gtest/ucp/test_ucp_am.cc

```diff
@@ -702,6 +704,68 @@ UCS_TEST_P(test_ucp_am_nbx_eager_memtype, basic)
 UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(test_ucp_am_nbx_eager_memtype)
 
 
+class test_ucp_am_nbx_eager_data_release : public test_ucp_am_nbx {
+public:
+    test_ucp_am_nbx_eager_data_release()
+    {
+        modify_config("RNDV_THRESH", "inf");
+        modify_config("ZCOPY_THRESH", "inf");
+        m_data_ptr = NULL;
+    }
+
+    virtual ucs_status_t
+    am_data_handler(const void *header, size_t header_length, void *data,
+                    size_t length, const ucp_am_recv_param_t *rx_param)
+    {
+        EXPECT_FALSE(m_am_received);
+        EXPECT_TRUE(rx_param->recv_attr & UCP_AM_RECV_ATTR_FLAG_DATA);
+
+        m_am_received = true;
+        m_data_ptr    = data;
```

Comment:
instead of adding new test class, can we extend the existing test_ucp_am_nbx so it could test different combinations of eager/rndv/data flags?

Reply:
`test_ucp_am_nbx ` already contains core functionality, other classes just reload its virtual functions and adds some specific to avoid mess by handling everything in one class

Reply:
so why the issue did not reproduce with test_ucp_am_nbx ?

Reply:
there were no tests for releasing deferred eager messages (tests were written for rndv only), here I add sthe tests

Reply:
any reason not to add these tests to base class?

Reply:
yes, because release flow is different for rendezvous and is tested separately. Keeping a compact class for testing certain functionality seems to be clear